components/eamxx/src/diagnostics/field_at_height.cpp

```diff
@@ -38,6 +38,12 @@ FieldAtHeight (const ekat::Comm& comm, const ekat::ParameterList& params)
  : AtmosphereDiagnostic(comm,params)
 {
   m_field_name = m_params.get<std::string>("field_name");
+  auto surf_ref = m_params.get<std::string>("surface_reference");
+  EKAT_REQUIRE_MSG(surf_ref == "sealevel" or surf_ref == "surface",
+      "Error! Invalid surface reference for FieldAtHeight.\n"
```

Comment:
I would also point out to the user what the valid options are. E.g.,

```suggestion
  auto surf_ref = m_params.get<std::string>("surface_reference");
    EKAT_REQUIRE_MSG(surf_ref == "sealevel" or surf_ref == "surface",
      "Error! Invalid surface reference for FieldAtHeight.\n"
     " -        field name: " + m_field_name + "\n"
     " - surface reference: " + surf_ref + "\n"
     " - valid options: sealevel, surface\n");
```